client/src/pages/course/score.tsx

```diff
@@ -130,6 +137,31 @@ export function Page(props: CoursePageProps) {
     loadInitialData();
   }, []);
 
+  const handleSettings = () => {
+    setIsVisibleSettings(!isVisibleSetting);
+  };
+
+  const handleModalCancel = () => {
+    setIsVisibleSettings(!isVisibleSetting);
+  };
+
+  const handleModalOk = (values: Store) => {
+    setIsNotVisibleColumns(JSON.stringify(Object.keys(values).filter((value: string) => values[value] === false)));
+    setIsVisibleSettings(!isVisibleSetting);
+  };
+
+  const getVisibleTasks = (tasks: CourseTask[]) => {
+    return tasks.map(task => {
+      notVisibleColumns?.includes(task.name) ? (task.isVisible = false) : (task.isVisible = true);
+      return task;
+    });
+  };
+
+  const getVisibleColumns = (columns: any) => {
```

Comment:
Really any? Or it is possible to define types for columns?

Reply:
done